### PR TITLE
[Refactor] Refactor network list

### PR DESCRIPTION
**Pull request by yzxiu**
Part of https://github.com/containerd/nerdctl/issues/1680

Checklist:

- [x]  Create a file in pkg/api/types/${cmd}_types.go, and define the CommandOption for this command
- [x]  Create some file in pkg/cmd/${cmd}, and move the command entry point in real into this package

Signed-off-by: yaozhenxiu <946666800@qq.com>

### CODE REVIEW

File: cmd/nerdctl/network_ls.go [hunk 1]
```diff
@@ -65,86 +50,14 @@ func networkLsAction(cmd *cobra.Command, args []string) error {
...
-			if p.ID != "" {
-				fmt.Fprintln(w, p.ID)
-			}
-		} else {
-			fmt.Fprintf(w, "%s\t%s\t%s\n", p.ID, p.Name, p.file)
-		}
-	}
-	if f, ok := w.(formatter.Flusher); ok {
-		return f.Flush()
+	options := &types.NetworkListCommandOptions{
```
Comment (djdongjin) [inline cmd/nerdctl/network_ls.go:57]: can you change to non-pointer return type?

File: cmd/nerdctl/network_ls.go [hunk 2]
```diff
@@ -65,86 +50,14 @@ func networkLsAction(cmd *cobra.Command, args []string) error {
...
+		GOptions: globalOptions,
+		Quiet:    quiet,
+		Format:   format,
 	}
-	return nil
+	return network.List(options, cmd.OutOrStdout())
```
Comment (djdongjin) [inline cmd/nerdctl/network_ls.go:62]: we probably should add a `context.Context` argument and pass in `cmd.Context()`, similar to

https://github.com/containerd/nerdctl/blob/cfe96a9d17b293463ad754ebdae6f0b933d7ae39/cmd/nerdctl/images.go#L133

**Review by djdongjin — CHANGES_REQUESTED** [review]

File: cmd/nerdctl/network_ls.go (hunk 1 above)
Comment (yzxiu) [reply cmd/nerdctl/network_ls.go]: done

File: cmd/nerdctl/network_ls.go (hunk 2 above)
Comment (yzxiu) [reply cmd/nerdctl/network_ls.go]: done

**Review by djdongjin — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**